http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -193,6 +161,58 @@ public boolean matches(String type, URI uri, String realm)
         return NEGOTIATE.equalsIgnoreCase(type);
     }
 
+    private void ensureClientCredentialUpdated()
+            throws LoginException, GSSException
+    {
+        if (clientCredential == null || clientCredential.getRemainingLifetime() <= MIN_CREDENTIAL_LIFE_TIME_SEC) {
+            if (loginContext != null) {
+                loginContext.logout();
```

Comment:
we do need a new login session to get updated TGT and thus clientCredential, this is also what OSS hadoop does, https://github.com/apache/hadoop/blob/773c670943757681feeafb227a2d0c29d48f38f1/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java#L1039-L1057.